core/src/main/scala/com/mesosphere/usi/core/logic/MesosEventsLogic.scala

```diff
@@ -1,90 +1,33 @@
 package com.mesosphere.usi.core.logic
 
-import com.mesosphere.ImplicitStrictLogging
-import com.mesosphere.LoggingArgs
 import java.time.Instant
 
+import com.mesosphere.{ImplicitStrictLogging, LoggingArgs}
 import com.mesosphere.mesos.client.MesosCalls
 import com.mesosphere.usi.core._
-import com.mesosphere.usi.core.matching.ResourceMatcher
+import com.mesosphere.usi.core.matching.{FCFSOfferMatcher, OfferMatcher}
 import com.mesosphere.usi.core.models._
-import com.mesosphere.usi.core.models.resources.{ResourceRequirement, ResourceType}
 import org.apache.mesos.v1.scheduler.Protos.{Call => MesosCall, Event => MesosEvent}
 import org.apache.mesos.v1.{Protos => Mesos}
+import SchedulerLogicHelpers._
 
-import scala.annotation.tailrec
 import scala.collection.JavaConverters._
 
 /**
   * The current home for USI Mesos event related logic
   */
-private[core] class MesosEventsLogic(mesosCallFactory: MesosCalls) extends ImplicitStrictLogging {
-  import SchedulerLogicHelpers._
-  private case class ResourceMatch(podSpec: PodSpec, resources: Seq[Mesos.Resource])
-  @tailrec private def maybeMatchPodSpec(
-      remainingResources: Map[ResourceType, Seq[Mesos.Resource]],
-      matchedResources: List[Mesos.Resource],
-      resourceRequirements: List[ResourceRequirement])
-    : Option[(List[Mesos.Resource], Map[ResourceType, Seq[Mesos.Resource]])] = {
-    resourceRequirements match {
-      case Nil =>
-        Some((matchedResources, remainingResources))
-      case req :: rest =>
-        ResourceMatcher.matchAndConsume(req, remainingResources.getOrElse(req.resourceType, Nil)) match {
-          case Some(matchResult) =>
-            maybeMatchPodSpec(
-              remainingResources.updated(req.resourceType, matchResult.remainingResource),
-              matchResult.matchedResources.toList ++ matchedResources,
-              rest)
-          case None =>
-            // we didn't match
-            None
-        }
-    }
-  }
-
-  @tailrec private def matchPodSpecsTaskRecords(
-      offer: Mesos.Offer,
-      remainingResources: Map[ResourceType, Seq[Mesos.Resource]],
-      result: Map[PodId, List[Mesos.TaskInfo]],
-      pendingLaunchPodSpecs: List[PodSpec]): Map[PodId, List[Mesos.TaskInfo]] = {
+private[core] class MesosEventsLogic(mesosCallFactory: MesosCalls, offerMatcher: OfferMatcher = new FCFSOfferMatcher())
+    extends ImplicitStrictLogging {
 
+  private[core] def matchOffer(offer: Mesos.Offer, specs: Iterable[PodSpec]): (Set[PodId], SchedulerEventsBuilder) = {
```

Comment:
can we find a better name for this method? since it's not just matching and now there's offer matcher interface doing the matching